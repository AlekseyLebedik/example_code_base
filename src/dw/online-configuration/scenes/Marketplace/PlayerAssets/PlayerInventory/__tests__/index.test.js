import React from 'react';
import { shallow } from 'enzyme';

import { PlayerItemsStateless } from '../index';

describe('Player Inventory', () => {
  const getPlayerItems = jest.fn();
  const onSelectRarityFilter = jest.fn();
  const onInventoryFilterChange = jest.fn();
  const onSelectItems = jest.fn();

  const renderContainer = (customProps = {}) => {
    const propsConnected = {
      getPlayerItems,
      onSelectRarityFilter,
      onInventoryFilterChange,
      onSelectItems,
      maxQuantity: 5,
      isClan: false,
      ...customProps,
    };

    return shallow(<PlayerItemsStateless {...propsConnected} />);
  };

  beforeEach(() => {
    getPlayerItems.mockClear();
  });

  describe('Connected Player Inventory', () => {
    it('renders default values', () => {
      const wrapper = renderContainer();
      expect(wrapper).toMatchSnapshot();
    });

    it('should not call getPlayerItems without userId or validContext', () => {
      renderContainer({ userId: undefined, validContext: undefined });
      expect(getPlayerItems).not.toHaveBeenCalled();
    });

    it('should call getPlayerItems when userId and valid context are present', () => {
      const userId = 'test1';
      const validContext = 'text_context';
      renderContainer({ userId, validContext });
      expect(getPlayerItems).toHaveBeenCalled();
    });

    it('should call getPlayerItems again when context changes', () => {
      const userId = 'test1';
      const validContext = 'text_context';
      const wrapper = renderContainer({ userId, validContext });
      expect(getPlayerItems).toHaveBeenCalledTimes(1);
      const changedContext = 'text_context2';
      wrapper.setProps({ validContext: changedContext });
      expect(getPlayerItems).toHaveBeenCalledTimes(2);
    });
  });
});
