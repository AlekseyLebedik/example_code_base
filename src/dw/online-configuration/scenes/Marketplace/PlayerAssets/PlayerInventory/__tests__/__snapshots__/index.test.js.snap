// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player Inventory Connected Player Inventory renders default values 1`] = `
<InventoryItemsStateless
  actionsRender={[Function]}
  checked={false}
  columnsDefs={
    Array [
      Object {
        "field": "quantity",
        "headerName": "Units",
        "sortable": true,
        "width": 150,
      },
    ]
  }
  darkMode={false}
  formatDate={[Function]}
  getPlayerItems={[MockFunction]}
  hasEditPermission={false}
  invalidItems={Array []}
  isClan={false}
  isLoading={false}
  items={Array []}
  maxQuantity={5}
  onInventoryFilterChange={[MockFunction]}
  onRemoveItems={null}
  onSelectItems={[MockFunction]}
  onSelectRarityFilter={[MockFunction]}
  searchInput=""
  selectedItems={Array []}
/>
`;
